# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/SocketProcessHost.cpp
# Commit: 10e01cf10faf
# Full Hash: 10e01cf10fafca4a150c2f76b3ce4e33bbd06223
# Author: Michael Froman <mfroman@mozilla.com>
# Date: 2020-03-09 21:52:54
# Regressor Bug: 1608558
# File Overlap Count: 1
# Description:
#   Bug 1608558 - pt1 - add linux sandboxing to socket process. r=gcp
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D60014
# ==============================================================================

diff -r 61e14505a488 -r 10e01cf10faf netwerk/ipc/SocketProcessHost.cpp
--- a/netwerk/ipc/SocketProcessHost.cpp	Fri Mar 06 20:29:47 2020 +0000
+++ b/netwerk/ipc/SocketProcessHost.cpp	Mon Mar 09 14:56:03 2020 +0000
@@ -9,6 +9,11 @@
 #include "nsIObserverService.h"
 #include "SocketProcessParent.h"
 
+#if defined(XP_LINUX) && defined(MOZ_SANDBOX)
+#  include "mozilla/SandboxBroker.h"
+#  include "mozilla/SandboxBrokerPolicyFactory.h"
+#endif
+
 #ifdef MOZ_GECKO_PROFILER
 #  include "ProfilerParent.h"
 #endif
@@ -189,6 +194,24 @@
     DebugOnly<nsresult> result = ioService->GetOffline(&offline);
     MOZ_ASSERT(NS_SUCCEEDED(result), "Failed getting offline?");
 
+    Maybe<FileDescriptor> brokerFd;
+
+#if defined(XP_LINUX) && defined(MOZ_SANDBOX)
+    // for now, use the UtilityPolicy like RDD does
+    auto policy =
+        SandboxBrokerPolicyFactory::GetUtilityPolicy(GetActor()->OtherPid());
+    if (policy != nullptr) {
+      brokerFd = Some(FileDescriptor());
+      mSandboxBroker = SandboxBroker::Create(
+          std::move(policy), GetActor()->OtherPid(), brokerFd.ref());
+      // This is unlikely to fail and probably indicates OS resource
+      // exhaustion.
+      Unused << NS_WARN_IF(mSandboxBroker == nullptr);
+      MOZ_ASSERT(brokerFd.ref().IsValid());
+    }
+    Unused << GetActor()->SendInitLinuxSandbox(brokerFd);
+#endif  // XP_LINUX && MOZ_SANDBOX
+
 #ifdef MOZ_GECKO_PROFILER
     Unused << GetActor()->SendInitProfiler(
         ProfilerParent::CreateForProcess(GetActor()->OtherPid()));